### PR TITLE
DAPR Container App

### LINKED ISSUE #131: [Module Proposal]: DAPR Container App
### Have you checked this module does not already exist in the registry?

Yes

### Why is the module needed?

To be able to add a DAPR optimised app onto a DAPR Container App Environment ( #118 )

### Module path

app/dapr-containerapp

### Describe the module

It will focus on the Microsoft.App/containerApps resource, setting DAPR config optimised for the component type patterns used in the #118 

It will optionally create a User Assigned Identity (and RBAC) that can pull a container image from a precreated ACR.

The 1.0.1 of the module will be modelled on the 2 most common DAPR application scenarios.

### PR DESCRIPTION
**Pull request by Gordonby**
## Description

Closes #131 

Adds a new module that can be leveraged with the dapr-containerapps-environment module to quickly enable developers creating their dapr infrastructure.

## Adding a new module

- [x] A proposal has been submitted and approved.
- [x] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [x] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.

### CODE REVIEW

**Comment by Gordonby** [comment]
The bicep design here is a little bit clunky because of https://github.com/Azure/bicep/issues/7836

See my note in main.bicep

File: modules/app/dapr-containerapp/main.bicep [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  memorySize: memorySize
+  minReplicas: minReplicas
+  revisionMode: revisionMode
+  revisionSuffix: revisionSuffix
+  tags: tags
+  targetPort: targetPort
+ }
+}
+
+module containerAppsACR 'containerapp.bicep' = if (!empty(azureContainerRegistry)) {
```
Comment (alex-frankel) [inline modules/app/dapr-containerapp/main.bicep:119]: isn't this supposed to point to `containerapp-acr.bicep`?

Comment (Gordonby) [reply modules/app/dapr-containerapp/main.bicep:119]: I'm glad someone's checking 😆

Comment (Gordonby) [reply modules/app/dapr-containerapp/main.bicep:119]: ok, fixed the problem and added a test to cover the capability.

Comment (alex-frankel) [reply modules/app/dapr-containerapp/main.bicep:119]: I'm not seeing a commit since this comment, but will keep an eye out for it

**Comment by Gordonby** [comment]
ok @alex-frankel - much better now.

i also made the managedId creation optional as that was bothering me.

**Review by alex-frankel — APPROVED** [review]

**Review by alex-frankel — APPROVED** [review]

**Pull request merged**